fem/plinearform.cpp

```diff
@@ -43,18 +43,9 @@ void ParLinearForm::MakeRef(ParFiniteElementSpace *pf, Vector &v, int v_offset)
    pfes = pf;
 }
 
-void ParLinearForm::Assemble(bool use_device)
+void ParLinearForm::Assemble()
 {
-   bool all_supports_device = use_device;
-
-   if (use_device)
-   {
-      bool supports_device = SupportsDevice();
-      MPI_Allreduce(&supports_device, &all_supports_device, 1,
```

Comment:
This logic was needed when `LinearForm::SupportsDevice()` did not return the same value in parallel (`face_info` related?). If we remove it here, we might need to move it inside `SupportsDevice`.

Reply:
This sounds like a bug to me, but if there is indeed a reason for `SupportsDevice()` to return different values on different MPI ranks, than I suggest we make it virtual and use this in the implementation of `ParLinearForm::SupportsDevice()`.

Reply:
@camierjs and @pazner, am I correct that `LinearForm::SupportsDevice()` can return different values on different MPI ranks because of the following code?

https://github.com/mfem/mfem/pull/3144/files#diff-c6385ef7dfee72c05f1246af1a530036f80e2cefb776734e0c03328530c56572R124-R141

If so, should I add `ParLinearForm::SupportsDevice()` as suggested above?

Reply:
See https://github.com/mfem/mfem/pull/3235/commits/6eb6556a048d768ffe6141bc58e16056177e2ae2